plugins/woocommerce/client/legacy/js/admin/meta-boxes-order.js

```diff
@@ -333,13 +333,25 @@ jQuery( function ( $ ) {
 			$( '#woocommerce-order-items' ).unblock();
 		},
 
+		filter_data: function( handle, data ) {
+			const filteredData = $( '#woocommerce-order-items' ).triggerHandler( `${handle}_ajax_data`, [ data ] );
```

Comment:
```suggestion
			const filteredData = $( '#woocommerce-order-items' ).triggerHandler( `woocommerce_order_meta_box_${handle}_ajax_data`, [ data ] );
```

It would be great to 'namespace' this: though it is bound to a WooCommerce element ID, there is still some potential for collisions with third party code, etc. This would be an easy way to reduce that.

Reply:
As you correctly recognized, the selector is already limiting the scope of the event name. Thus, the chances of a conflict should be negligible. Furthermore, your suggestion would not be consistent with how the event name is defined in `meta-boxes-product-variation.js`, which is the only other occurrence of `triggerHandler` used in a similar fashion:

https://github.com/woocommerce/woocommerce/blob/3669426bd6af4a6816184f843ffdaf9790af8d84/plugins/woocommerce/client/legacy/js/admin/meta-boxes-product-variation.js#L814

In that script, the variable part of the event name is not namespaced either, most probably for the same reason of the selector restricting the scope of the event itself.

Reply:
> The selector is already limiting the scope of the event name.

Only to a degree. For instance, consider something like this:

`$( div ).trigger( 'delete_item_ajax_data', payload )`

> In that script, the variable part of the event name is not namespaced either

Sure, but while consistency is important it doesn't mean we can't break with a historic pattern in order to improve elsewhere.

Reply:
> ```$( div ).trigger( 'delete_item_ajax_data', payload )```

Well, that line seems to be written by someone who is looking for trouble! 😛 

> Sure, but while consistency is important it doesn't mean we can't break with a historic pattern in order to improve elsewhere.

You make a good point, here. 👍

Reply:
Hah, well ... I'm definitely not suggesting it's a model of good coding; but we do exist in a large and varied ecosystem (even the best products may contain oversights, there are some site owners may implement code on a copy-and-paste basis, and so on). A small and reasonable precaution can help to mitigate these things and reduce future pain.